packages/flutter/lib/src/material/chip.dart

```diff
@@ -992,13 +992,21 @@ class _RawChipState extends State<RawChip> with MaterialStateMixin, TickerProvid
 
   OutlinedBorder _getShape(ThemeData theme, ChipThemeData chipTheme, ChipThemeData chipDefaults) {
     final BorderSide? resolvedSide = MaterialStateProperty.resolveAs<BorderSide?>(widget.side, materialStates)
-      ?? MaterialStateProperty.resolveAs<BorderSide?>(chipTheme.side, materialStates)
-      ?? MaterialStateProperty.resolveAs<BorderSide?>(chipDefaults.side, materialStates);
+      ?? MaterialStateProperty.resolveAs<BorderSide?>(chipTheme.side, materialStates);
     final OutlinedBorder resolvedShape = MaterialStateProperty.resolveAs<OutlinedBorder?>(widget.shape, materialStates)
       ?? MaterialStateProperty.resolveAs<OutlinedBorder?>(chipTheme.shape, materialStates)
       ?? MaterialStateProperty.resolveAs<OutlinedBorder?>(chipDefaults.shape, materialStates)
+      // TODO(tahatesser): Remove this fallback when Material 2 is deprecated.
       ?? const StadiumBorder();
-    return resolvedShape.copyWith(side: resolvedSide);
+    // If the side is provided, shape uses the provided side.
+    if (resolvedSide != null) {
+      return resolvedShape.copyWith(side: resolvedSide);
+    }
+    // If the side is not provided and the shape's side is not [BorderSide.none],
+    // then the shape's side is used. Otherwise, the default side is used.
+    return resolvedShape.side != BorderSide.none
```

Comment:
So if I want to specify a shape with no border, then I need to specify both the `shape` and the `side` (as BorderSide.none) because specifying a shape with `BorderSide.none` will cause us to use the default side. I guess that's OK, since the OutlinedBorder classes use `BorderSide.none` by default. It's a little confusing.  A separate PR that clarified this in the API doc might help.